datumaro/plugins/data_formats/datumaro_binary/exporter.py

```diff
@@ -44,20 +46,31 @@ def _dump_encryption_field(self) -> int:
 
         return self._fp.write(struct.pack(f"I{len(msg)}s", len(msg), msg))
 
-    def _dump_header(self, header: Any):
-        msg = DictMapper.forward(header)
+    def _dump_header(self, header: Any, elem_type: type):
+        if elem_type == str:
+            msg = StringMapper.forward(header)
+        elif elem_type == dict:
+            msg = DictMapper.forward(header)
+        else:
+            raise DatasetExportError(f"unsupported element type for binary format: {elem_type}")
 
         if self._crypter.key is not None:
             msg = self._crypter.encrypt(msg)
 
         length = struct.pack("I", len(msg))
         return self._fp.write(length + msg)
 
+    def _dump_version(self):
+        self._dump_header(DATUMARO_FORMAT_VERSION, str)
```

Comment:
```suggestion
        self._dump_header({"dm_format_version": DATUMARO_FORMAT_VERSION})
```

Reply:
Please dump it with `dict` as same as `media_type` not to increase code branching.